### ISSUE TITLE
Boost extraction extremely slow on Windows

### ISSUE THREAD

**Issue by magnific0** [Tudat/tudatBundle#10]
Currently the tar.bz2 archive of boost is downloaded and extracted using CMake's build-in tar function (```cmake -E tar xvf```). On Windows this process takes much longer, compared to Linux or Mac.

It should be investigated whether a different archive (boost has zip, tar.gz and 7z versions too) or process be used to extract files on Windows instead.

The process takes about 6 minutes on an Intel i7 Haswell @ 2.30GHz!

**Comment by magnific0** [comment]
Okay I just did a benchmark of the several file formats using ```cmake -E tar xf``` and alternative commands and they are all terrible.

    cmake -E tar xf boost_1_63_0.zip          Windows        273.73s        142.7MB
    cmake -E tar xf boost_1_63_0.7z           Windows        303.46s         71.5MB
    cmake -E tar xf boost_1_63_0.tar.gz       Windows        276.17s         97.5MB
    cmake -E tar xf boost_1_63_0.tar.bz2      Windows        288.77s         82.0MB
    7z.exe x boost_1_63_0.7z                  Windows        288.75s
    powershell.exe [...]                      Windows        250.29s
    7z x boost_1_63_0.7z                      Linux            9.753s
    unzip boost_1_63_0.zip                    Linux            5.036s
    tar -xf boost_1_63_0.tar.gz               Linux            2.870s

It is evident that for Windows the difference in filesize of the different archives is more significant than the differences in extraction time. The extracted size is 670MB containing 53661 files and 4199 folders.

In conclusion, performance on Windows is just awful and there is nothing we can do about it. It very likely is caused by the large number of files.

**Comment by MaxBarraclough** [comment]
Strictly speaking there *is* another way: distribute the Boost archive in .iso format and simply mount the .iso in Windows as a read-only drive (which Windows supports natively). This way you never have to extract the archive, avoiding having to create 65,000 small files. (As you say, this is oddly *much* slower in Windows than in Linux.)

Of course, this approach will strike the average Windows developer as being highly irregular, which of course it is, but I've tried it out and it does appear to work. Just have to be careful to use the right parameters so that the .iso can cope with long paths and deep nesting.

**Comment by auxeon** [comment]
I also tried another approach of using wsl on windows 10, I speculate this has something to do with the huge number of small files themselves and how the windows filesystem works. 

this is because majority of the time is spent on in sys calls and waiting for stuff to execute - the user space processing is about 4.5s

these were the time results i got

```shell
... 
  inflating: boost_1_74_0/libs/ratio/example/si_physics.cpp
  inflating: boost_1_74_0/libs/ratio/example/duration.hpp
  inflating: boost_1_74_0/libs/ratio/example/display_ex.cpp
  inflating: boost_1_74_0/libs/ratio/index.html

real    7m39.851s
user    0m4.531s
sys     4m23.234s
```
Processor 
```
Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
```
